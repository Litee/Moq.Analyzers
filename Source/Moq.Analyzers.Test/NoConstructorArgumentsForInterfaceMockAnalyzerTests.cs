using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.CodeAnalysis.Diagnostics;
using System.IO;
using TestHelper;
using Xunit;

namespace Moq.Analyzers.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class NoConstructorArgumentsForInterfaceMockAnalyzerTests : DiagnosticVerifier
    {

        [Fact]
        public void ShouldFailIfMockedInterfaceHasParameters()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/NoConstructorArgumentsForInterfaceMock_1.cs")));
        }

        [Fact]
        public void ShouldPassIfCustomMockClassIsUsed()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/NoConstructorArgumentsForInterfaceMock_2.cs")));
        }

        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
        {
            return new NoConstructorArgumentsForInterfaceMockAnalyzer();
        }
    }
}