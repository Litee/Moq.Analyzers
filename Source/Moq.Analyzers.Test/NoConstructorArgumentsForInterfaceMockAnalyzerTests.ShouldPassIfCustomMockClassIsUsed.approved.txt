Diagnostic 1
	Id: Moq1001
	Location: SourceFile(Test0.cs[612..621))
	Highlight: (1, true)
	Lines: var mock1 = new Moq.Mock<IMyService>(1, true);
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 2
	Id: Moq1001
	Location: SourceFile(Test0.cs[713..718))
	Highlight: ("2")
	Lines: var mock2 = new Moq.Mock<NoConstructorArgumentsForInterfaceMock_2.IMyService>("2");
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 3
	Id: Moq1001
	Location: SourceFile(Test0.cs[769..800))
	Highlight: (Moq.MockBehavior.Default, "3")
	Lines: var mock3 = new Moq.Mock<IMyService>(Moq.MockBehavior.Default, "3");
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 4
	Id: Moq1001
	Location: SourceFile(Test0.cs[892..921))
	Highlight: (MockBehavior.Loose, 4, true)
	Lines: var mock4 = new Moq.Mock<NoConstructorArgumentsForInterfaceMock_2.IMyService>(MockBehavior.Loose, 4, true);
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 5
	Id: Moq1001
	Location: SourceFile(Test0.cs[972..994))
	Highlight: (MockBehavior.Default)
	Lines: var mock5 = new Moq.Mock<IMyService>(MockBehavior.Default);
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

Diagnostic 6
	Id: Moq1001
	Location: SourceFile(Test0.cs[1086..1108))
	Highlight: (MockBehavior.Default)
	Lines: var mock6 = new Moq.Mock<NoConstructorArgumentsForInterfaceMock_2.IMyService>(MockBehavior.Default);
	Severity: Warning
	Message: Do not specify parameters for mocked interface.

